.github/workflows/checks.yaml

```diff
@@ -71,33 +71,32 @@ jobs:
       - name: target
         run: "rustc -vV | sed -n 's|host: ||p'"
         if: matrix.os != 'windows-2022'
-      - name: Install cargo-llvm-cov
-        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
-        with:
-          tool: cargo-llvm-cov
-      - name: Install cargo-nextest
-        uses: taiki-e/install-action@56ab7930c591507f833cbaed864d201386d518a8
+      - name: Install cargo-llvm-cov & cargo-nextest
+        uses: taiki-e/install-action@a48a50298f98c47e46a957ae6f82c44cc4878e42
         with:
-          tool: cargo-nextest
+          tool: cargo-llvm-cov,cargo-nextest
       - name: cargo build
         run: cargo build --target ${{matrix.target}} ${{ matrix.features }}
-      - name: cargo nextest # reports segfaults in a helpful way
-        run: cargo nextest run --target ${{matrix.target}} ${{ matrix.features }} --no-fail-fast
+      - name: cargo llvm-cov nextest
+        if: matrix.os != 'windows-2022'
+        run: cargo llvm-cov nextest --target ${{matrix.target}} ${{ matrix.features }} --lcov --output-path lcov.info
         env:
           RUST_BACKTRACE: 1
```

Comment:
we can use `cargo llvm-cov nextest` to run tests with cargo-nextest, and gather the coverage information at the same time. That saves a build + test run, making this action a bit faster.